app/src/main/java/org/p2p/wallet/home/ui/new/NewSelectTokenPresenter.kt

```diff
@@ -9,17 +14,37 @@ import org.p2p.wallet.home.model.SelectableTokenRoundedState.BOTTOM_ROUNDED
 import org.p2p.wallet.home.model.SelectableTokenRoundedState.NOT_ROUNDED
 import org.p2p.wallet.home.model.SelectableTokenRoundedState.ROUNDED
 import org.p2p.wallet.home.model.SelectableTokenRoundedState.TOP_ROUNDED
-import org.p2p.core.token.Token
+import org.p2p.wallet.home.repository.UserTokensRepository
 
-class NewSelectTokenPresenter : BasePresenter<NewSelectTokenContract.View>(), NewSelectTokenContract.Presenter {
+class NewSelectTokenPresenter(
+    private val userTokensRepository: UserTokensRepository,
+    private val selectedTokenMintAddress: String?,
+    private val selectableTokens: List<Token.Active>?
+) : BasePresenter<NewSelectTokenContract.View>(), NewSelectTokenContract.Presenter {
 
-    private val mappedTokens = mutableListOf<SelectTokenItem>()
+    private var mappedTokens: List<SelectTokenItem> by observable(emptyList()) { _, _, new ->
+        view?.showTokens(new)
+        view?.showEmptyState(isVisible = new.isEmpty())
+    }
 
-    override fun load(tokens: List<Token.Active>, selectedToken: Token.Active?) {
-        mapInitialTokens(selectedToken, tokens)
+    override fun attach(view: NewSelectTokenContract.View) {
+        super.attach(view)
+
+        if (selectableTokens != null) {
```

Comment:
You can use 
`if (selectableTokens != null) return`
and drop out `else` statement

Reply:
how can i drop else statement if there is logic inside? 
refactored this stuff anyway, you can check